{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Manipulation of PEAKS de novo results of _T. weiss_  LC-MS/MS data using Python.\n",
    "\n",
    "Starting with:\n",
    "\n",
    "    PEAKS de novo results (.csv) of PTM-optimized sequencing >80% ALC\n",
    "    from Thermo Fusion tribrid runs at the UW Proteomics Resource center, January 2021\n",
    "    combined from multiple injections\n",
    "\n",
    "Goal:\n",
    "\n",
    "    Files with stripped (no PTMs) peptide lists and\n",
    "    Columns with #'s of each modification in every sequence\n",
    "    Column with stripped peptide lengths (# amino acids)\n",
    "    \n",
    "### To use:\n",
    "\n",
    "#### 1. Change the input file name in *IN 4*\n",
    "#### 2. Change output file name in *IN 6*, *IN 7*, *IN 8*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/millieginty/Documents/git-repos/rot-mayer/data/MED_Weissrot_Fusion_UWPR2021/MED_Weissrot_Fusion_332-T12nd-all_DENOVO_100\n"
     ]
    }
   ],
   "source": [
    "cd /home/millieginty/Documents/git-repos/rot-mayer/data/MED_Weissrot_Fusion_UWPR2021/MED_Weissrot_Fusion_332-T12nd-all_DENOVO_100/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# redundant Peaks peptides >80% ALC in combined dataframe: 2713\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Fraction</th>\n",
       "      <th>Scan</th>\n",
       "      <th>Source File</th>\n",
       "      <th>Peptide</th>\n",
       "      <th>Tag Length</th>\n",
       "      <th>ALC (%)</th>\n",
       "      <th>length</th>\n",
       "      <th>m/z</th>\n",
       "      <th>z</th>\n",
       "      <th>RT</th>\n",
       "      <th>Area</th>\n",
       "      <th>Mass</th>\n",
       "      <th>ppm</th>\n",
       "      <th>PTM</th>\n",
       "      <th>local confidence (%)</th>\n",
       "      <th>tag (&gt;=0%)</th>\n",
       "      <th>mode</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>22</td>\n",
       "      <td>14158</td>\n",
       "      <td>20210115_Weissrot_332_T12_nodigest_DDA_120min_...</td>\n",
       "      <td>HEDWQLK</td>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>478.2356</td>\n",
       "      <td>2</td>\n",
       "      <td>48.27</td>\n",
       "      <td>2130000.0</td>\n",
       "      <td>954.4559</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98 100 99 98 95 100 99</td>\n",
       "      <td>HEDWQLK</td>\n",
       "      <td>HCD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>22</td>\n",
       "      <td>25708</td>\n",
       "      <td>20210115_Weissrot_332_T12_nodigest_DDA_120min_...</td>\n",
       "      <td>LQNLHLL</td>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>425.7617</td>\n",
       "      <td>2</td>\n",
       "      <td>85.28</td>\n",
       "      <td>411000.0</td>\n",
       "      <td>849.5072</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>96 97 98 100 100 100 100</td>\n",
       "      <td>LQNLHLL</td>\n",
       "      <td>HCD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>22</td>\n",
       "      <td>6806</td>\n",
       "      <td>20210115_Weissrot_332_T12_nodigest_DDA_120min_...</td>\n",
       "      <td>KR(+15.99)LVVEN</td>\n",
       "      <td>7</td>\n",
       "      <td>98</td>\n",
       "      <td>7</td>\n",
       "      <td>437.2560</td>\n",
       "      <td>2</td>\n",
       "      <td>25.49</td>\n",
       "      <td>979000.0</td>\n",
       "      <td>872.5079</td>\n",
       "      <td>-12.1</td>\n",
       "      <td>Oxidation(KPRY)</td>\n",
       "      <td>98 99 100 99 95 99 99</td>\n",
       "      <td>KR(+15.99)LVVEN</td>\n",
       "      <td>HCD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>22</td>\n",
       "      <td>26403</td>\n",
       "      <td>20210115_Weissrot_332_T12_nodigest_DDA_120min_...</td>\n",
       "      <td>DPN(+.98)LPLK(+42.01)H</td>\n",
       "      <td>8</td>\n",
       "      <td>98</td>\n",
       "      <td>8</td>\n",
       "      <td>488.7589</td>\n",
       "      <td>2</td>\n",
       "      <td>87.33</td>\n",
       "      <td>12100000.0</td>\n",
       "      <td>975.5025</td>\n",
       "      <td>0.7</td>\n",
       "      <td>Deamidation(N); Acetylation (K)</td>\n",
       "      <td>93 99 100 100 98 99 99 97</td>\n",
       "      <td>DPN(+.98)LPLK(+42.01)H</td>\n",
       "      <td>HCD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>22</td>\n",
       "      <td>8756</td>\n",
       "      <td>20210115_Weissrot_332_T12_nodigest_DDA_120min_...</td>\n",
       "      <td>KDETTLVDGH</td>\n",
       "      <td>10</td>\n",
       "      <td>97</td>\n",
       "      <td>10</td>\n",
       "      <td>557.7729</td>\n",
       "      <td>2</td>\n",
       "      <td>31.10</td>\n",
       "      <td>1210000.0</td>\n",
       "      <td>1113.5303</td>\n",
       "      <td>0.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>98 99 100 99 98 100 98 99 96 92</td>\n",
       "      <td>KDETTLVDGH</td>\n",
       "      <td>HCD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Fraction   Scan                                        Source File  \\\n",
       "0        22  14158  20210115_Weissrot_332_T12_nodigest_DDA_120min_...   \n",
       "1        22  25708  20210115_Weissrot_332_T12_nodigest_DDA_120min_...   \n",
       "2        22   6806  20210115_Weissrot_332_T12_nodigest_DDA_120min_...   \n",
       "3        22  26403  20210115_Weissrot_332_T12_nodigest_DDA_120min_...   \n",
       "4        22   8756  20210115_Weissrot_332_T12_nodigest_DDA_120min_...   \n",
       "\n",
       "                  Peptide  Tag Length  ALC (%)  length       m/z  z     RT  \\\n",
       "0                 HEDWQLK           7       98       7  478.2356  2  48.27   \n",
       "1                 LQNLHLL           7       98       7  425.7617  2  85.28   \n",
       "2         KR(+15.99)LVVEN           7       98       7  437.2560  2  25.49   \n",
       "3  DPN(+.98)LPLK(+42.01)H           8       98       8  488.7589  2  87.33   \n",
       "4              KDETTLVDGH          10       97      10  557.7729  2  31.10   \n",
       "\n",
       "         Area       Mass   ppm                              PTM  \\\n",
       "0   2130000.0   954.4559   0.7                              NaN   \n",
       "1    411000.0   849.5072   2.0                              NaN   \n",
       "2    979000.0   872.5079 -12.1                  Oxidation(KPRY)   \n",
       "3  12100000.0   975.5025   0.7  Deamidation(N); Acetylation (K)   \n",
       "4   1210000.0  1113.5303   0.9                              NaN   \n",
       "\n",
       "              local confidence (%)              tag (>=0%) mode  \n",
       "0           98 100 99 98 95 100 99                 HEDWQLK  HCD  \n",
       "1         96 97 98 100 100 100 100                 LQNLHLL  HCD  \n",
       "2            98 99 100 99 95 99 99         KR(+15.99)LVVEN  HCD  \n",
       "3        93 99 100 100 98 99 99 97  DPN(+.98)LPLK(+42.01)H  HCD  \n",
       "4  98 99 100 99 98 100 98 99 96 92              KDETTLVDGH  HCD  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read the CSVs into a dataframe using the read_csv function and call 'peaks'\n",
    "\n",
    "peaks332 = pd.read_csv(\"/home/millieginty/Documents/git-repos/rot-mayer/data/MED_Weissrot_Fusion_UWPR2021/MED_Weissrot_Fusion_332-T12nd-all_DENOVO_100/332-T12nd-all_DENOVO_100_DN.csv\")\n",
    "\n",
    "print(\"# redundant Peaks peptides >80% ALC in combined dataframe:\", len(peaks332))\n",
    "\n",
    "# look at the dataframe\n",
    "peaks332.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The peptide column has the masses of modifications (e.g., 57.02 Da for carbamidomethylation of cysteine). We want to make new columns with all that information and make a column with only the 'stripped' peptide sequence that's just amino acids - this we can then align against other sequences, for example.\n",
    "\n",
    "Modified residues were allowed for:\n",
    "\n",
    "    fixed carbamodimethylation of cysteine 57.021464 C\n",
    "    varialbe oxidation of methionine, lysine, proline, arginine, tyrosine: 15.9949 MKPRY\n",
    "    variable deamidation of asparagine, glumatine: 0.984016 NQ\n",
    "    variable methylation of lysine and arginine: 14.015650 KR\n",
    "    variable pyro-glu from glutamine: -17.03 Q\n",
    "    variable acetylation of lysine: 42.01 K\n",
    "\n",
    "\n",
    "We'll then write this manipulated dataframe to a new file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# use a count function to enumerate the # of A's (alanines) in each peptide\n",
    "peaks332['A'] = peaks332['Peptide'].str.count(\"A\")\n",
    "\n",
    "# use a count function to enumerate the # of C's (cysteines) in each peptide\n",
    "peaks332['C'] = peaks332['Peptide'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of D's (aspartic acids) in each peptide\n",
    "peaks332['D'] = peaks332['Peptide'].str.count(\"D\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (glutamic acids) in each peptide\n",
    "peaks332['E'] = peaks332['Peptide'].str.count(\"E\")\n",
    "\n",
    "# use a count function to enumerate the # of F's (phenylalanines) in each peptide\n",
    "peaks332['F'] = peaks332['Peptide'].str.count(\"F\")\n",
    "\n",
    "# use a count function to enumerate the # of G's (glycines) in each peptide\n",
    "peaks332['G'] = peaks332['Peptide'].str.count(\"G\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (histidines) in each peptide\n",
    "peaks332['H'] = peaks332['Peptide'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of I's (isoleucines) in each peptide\n",
    "# in peaks332 output, there will be no isoleucines (they're lumped in with leucines)\n",
    "peaks332['I'] = peaks332['Peptide'].str.count(\"I\")\n",
    "\n",
    "# use a count function to enumerate the # of K's (lysines) in each peptide\n",
    "peaks332['K'] = peaks332['Peptide'].str.count(\"K\")\n",
    "\n",
    "# use a count function to enumerate the # of L's (leucines) in each peptide\n",
    "# also these include the isoleucines\n",
    "peaks332['L'] = peaks332['Peptide'].str.count(\"L\")\n",
    "\n",
    "# use a count function to enumerate the # of M's (methionines) in each peptide\n",
    "peaks332['M'] = peaks332['Peptide'].str.count(\"M\")\n",
    "\n",
    "# use a count function to enumerate the # of N's (asparagines) in each peptide\n",
    "peaks332['N'] = peaks332['Peptide'].str.count(\"N\")\n",
    "\n",
    "# use a count function to enumerate the # of P's ([prolines]) in each peptide\n",
    "peaks332['P'] = peaks332['Peptide'].str.count(\"P\")\n",
    "\n",
    "# use a count function to enumerate the # of Q's (glutamines) in each peptide\n",
    "peaks332['Q'] = peaks332['Peptide'].str.count(\"Q\")\n",
    "\n",
    "# use a count function to enumerate the # of R's (arginines) in each peptide\n",
    "peaks332['R'] = peaks332['Peptide'].str.count(\"R\")\n",
    "\n",
    "# use a count function to enumerate the # of S's (serines) in each peptide\n",
    "peaks332['S'] = peaks332['Peptide'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of T's (threonines) in each peptide\n",
    "peaks332['T'] = peaks332['Peptide'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of V's (valines) in each peptide\n",
    "peaks332['V'] = peaks332['Peptide'].str.count(\"V\")\n",
    "\n",
    "# use a count function to enumerate the # of W's (tryptophans) in each peptide\n",
    "peaks332['W'] = peaks332['Peptide'].str.count(\"W\")\n",
    "\n",
    "# use a count function to enumerate the # of Y's (tyrosines) in each peptide\n",
    "peaks332['Y'] = peaks332['Peptide'].str.count(\"Y\")\n",
    "\n",
    "# use a count function to enumerate the # of carbamidomethylated C's in each peptide\n",
    "peaks332['c-carb'] = peaks332['Peptide'].str.count(\"57.02\")\n",
    "\n",
    "# use a count function to enumerate the # of oxidized M's in each peptide\n",
    "peaks332['m-oxid'] = peaks332['Peptide'].apply(lambda x: x.count('M(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized K's in each peptide\n",
    "peaks332['k-oxid'] = peaks332['Peptide'].apply(lambda x: x.count('K(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized P's in each peptide\n",
    "peaks332['p-oxid'] = peaks332['Peptide'].apply(lambda x: x.count('P(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized R's in each peptide\n",
    "peaks332['r-oxid'] = peaks332['Peptide'].apply(lambda x: x.count('R(+15.99)'))\n",
    "\n",
    "# use a count function to enumerate the # of oxidized Y's in each peptide\n",
    "peaks332['y-oxid'] = peaks332['Peptide'].apply(lambda x: x.count('Y(+15.99)'))\n",
    "\n",
    "# use a lamba function to enumerate the # of deamidated N's in each peptide\n",
    "peaks332['n-deam'] = peaks332['Peptide'].apply(lambda x: x.count('N(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of deamidated Q's in each peptide\n",
    "#peaks332['q-deam'] = peaks332['Peptide'].apply(lambda x: x.count('Q(+.98)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated K's in each peptide\n",
    "peaks332['k-meth'] = peaks332['Peptide'].apply(lambda x: x.count('K(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of methylated R's in each peptide\n",
    "peaks332['r-meth'] = peaks332['Peptide'].apply(lambda x: x.count('R(+14.02)'))\n",
    "\n",
    "# use a count function to enumerate the # of pyro glu Q's in each peptide\n",
    "peaks332['q-pyro'] = peaks332['Peptide'].apply(lambda x: x.count('Q(-17.03)'))\n",
    "\n",
    "# use a count function to enumerate the # of acetylation of K's in each peptide\n",
    "peaks332['k-acet'] = peaks332['Peptide'].apply(lambda x: x.count('K(+42.01)'))\n",
    "\n",
    "# create a column with 'stripped' peptide sequences using strip\n",
    "peaks332['stripped peptide'] = peaks332['Peptide'].str.replace(r\"\\(.*\\)\",\"\")\n",
    "\n",
    "# add a column with the stripped peptide length (number of AAs)\n",
    "peaks332['stripped length'] = peaks332['stripped peptide'].apply(len)\n",
    "\n",
    "# total the number of modifications in sequence\n",
    "peaks332['ptm-total'] = peaks332['c-carb'] + peaks332['m-oxid'] + peaks332['k-oxid'] + peaks332['p-oxid'] \\\n",
    "+ peaks332['r-oxid'] + peaks332['y-oxid'] + peaks332['n-deam'] + peaks332['k-meth'] + peaks332['r-meth'] \\\n",
    "+ peaks332['q-pyro'] + peaks332['k-acet']\n",
    "\n",
    "# calculate NAAF numerator for each peptide k\n",
    "peaks332['NAAF num.'] = peaks332['Area'] / peaks332['stripped length']\n",
    "\n",
    "# write modified dataframe to new txt file, same name + 'stripped'\n",
    "peaks332.to_csv(\"/home/millieginty/Documents/git-repos/rot-mayer/data/processed/PeaksDN/TW_332_T12_undigested_combine_PTMopt_DN80.csv\")\n",
    "\n",
    "# check out the results\n",
    "peaks332.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# made a new dataframe that contains the sums of certain columns \n",
    "# in the stripped peptide dataframe above (for >80% ALC)\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'A': peaks332['A'].sum(),\n",
    "        'C': peaks332['C'].sum(),\n",
    "        'D': peaks332['D'].sum(),\n",
    "        'E': peaks332['E'].sum(),\n",
    "        'F': peaks332['F'].sum(),\n",
    "        'G': peaks332['G'].sum(),\n",
    "        'H': peaks332['H'].sum(),\n",
    "        'I': peaks332['I'].sum(),\n",
    "        'K': peaks332['K'].sum(),\n",
    "        'L': peaks332['L'].sum(),\n",
    "        'M': peaks332['M'].sum(),\n",
    "        'N': peaks332['N'].sum(),\n",
    "        'P': peaks332['P'].sum(),\n",
    "        'Q': peaks332['Q'].sum(),\n",
    "        'R': peaks332['R'].sum(),\n",
    "        'S': peaks332['S'].sum(),\n",
    "        'T': peaks332['T'].sum(),\n",
    "        'V': peaks332['V'].sum(),\n",
    "        'W': peaks332['W'].sum(),\n",
    "        'Y': peaks332['Y'].sum(),\n",
    "        'c-carb': peaks332['c-carb'].sum(),\n",
    "        'm-oxid': peaks332['m-oxid'].sum(),\n",
    "        'k-oxid': peaks332['k-oxid'].sum(),\n",
    "        'p-oxid': peaks332['p-oxid'].sum(),\n",
    "        'r-oxid': peaks332['r-oxid'].sum(),\n",
    "        'y-oxid': peaks332['y-oxid'].sum(),\n",
    "        'n-deam': peaks332['n-deam'].sum(),\n",
    "        'k-meth': peaks332['k-meth'].sum(),\n",
    "        'r-meth': peaks332['r-meth'].sum(),\n",
    "        'q-pyro': peaks332['q-pyro'].sum(),\n",
    "        'k-acet': peaks332['k-acet'].sum(),\n",
    "        'Total area': peaks332['Area'].sum(),\n",
    "        'Total length': peaks332['stripped length'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks332 = pd.DataFrame(data, columns=['A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', 'L', \\\n",
    "                                            'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', \\\n",
    "                                            'c-carb', 'm-oxid', 'k-oxid', 'p-oxid', 'r-oxid', \\\n",
    "                                            'y-oxid', 'n-deam', 'k-meth', 'r-meth', 'q-pyro', \\\n",
    "                                            'k-acet', 'Total area', 'Total length'], index=index)\n",
    "\n",
    "# calculate percentage of C's with carb (should be 1.0)\n",
    "totalpeaks332['% C w/ carb'] = totalpeaks332['c-carb'] / totalpeaks332['C'] \n",
    "\n",
    "# calculate percentage of M's that are oxidized\n",
    "totalpeaks332['% M w/ oxid'] = totalpeaks332['m-oxid'] / totalpeaks332['M'] \n",
    "\n",
    "# calculate percentage of K's that are oxidized\n",
    "totalpeaks332['% K w/ oxid'] = totalpeaks332['k-oxid'] / totalpeaks332['K'] \n",
    "\n",
    "# calculate percentage of P's that are oxidized\n",
    "totalpeaks332['% P w/ oxid'] = totalpeaks332['p-oxid'] / totalpeaks332['P'] \n",
    "\n",
    "# calculate percentage of R's that are oxidized\n",
    "totalpeaks332['% R w/ oxid'] = totalpeaks332['p-oxid'] / totalpeaks332['R'] \n",
    "\n",
    "# calculate percentage of Y's that are oxidized\n",
    "totalpeaks332['% Y w/ oxid'] = totalpeaks332['y-oxid'] / totalpeaks332['Y'] \n",
    "\n",
    "# calculate percentage of N's that are deamidated\n",
    "totalpeaks332['% N w/ deam'] = totalpeaks332['n-deam'] / totalpeaks332['N'] \n",
    "\n",
    "# calculate percentage of K's that are methylated\n",
    "totalpeaks332['% K w/ meth'] = totalpeaks332['k-meth'] / totalpeaks332['K'] \n",
    "\n",
    "# calculate percentage of R's that are methylated\n",
    "totalpeaks332['% R w/ meth'] = totalpeaks332['r-meth'] / totalpeaks332['R'] \n",
    "\n",
    "# calculate percentage of Q's that are pyro glu'd\n",
    "totalpeaks332['% Q w/ pyro'] = totalpeaks332['q-pyro'] / totalpeaks332['Q'] \n",
    "\n",
    "# calculate percentage of K's that are acetylation\n",
    "totalpeaks332['% K w/ acet'] = totalpeaks332['k-acet'] / totalpeaks332['K'] \n",
    "\n",
    "# calculate NAAF denominator for all peptides in dataset i\n",
    "totalpeaks332['NAAF denom.'] = totalpeaks332['Total area'] / totalpeaks332['Total length']\n",
    "\n",
    "# write modified dataframe to new txt file\n",
    "totalpeaks332.to_csv(\"/home/millieginty/Documents/git-repos/rot-mayer/data/processed/PeaksDN/TW_332_T12_undigested_combine_PTMopt_DN80_totals.csv\")\n",
    "\n",
    "totalpeaks332.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# use the calculated NAAF factor (in totalpeaks dataframe, above) to caluclate the NAAF \n",
    "# NAAF: normalized normalized area abundance factor\n",
    "\n",
    "# don't have to worry here about DECOY hits messing with Area totals\n",
    "# but we would with Comet results\n",
    "\n",
    "NAAF80 = 243531.178471\n",
    "\n",
    "# use NAAF >80% ALC to get NAAF factor\n",
    "peaks332['NAAF factor'] = (peaks332['NAAF num.'])/NAAF80\n",
    "\n",
    "# make a dataframe that contains only what we need: sequences, AAs, PTMs\n",
    "peaksAAPTM_332 = peaks332[['stripped peptide', 'NAAF factor', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'K', 'I', 'L', \\\n",
    "                                'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', 'W', 'Y', 'c-carb', 'm-oxid', \\\n",
    "                                'k-oxid', 'p-oxid', 'r-oxid', 'y-oxid', 'n-deam', 'k-meth', 'r-meth', \\\n",
    "                                'q-pyro', 'k-acet']].copy()\n",
    "\n",
    "# multiply the NAAF80 factor by the AA total to normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaksAAPTM_332['A-NAAF80'] = peaksAAPTM_332['A'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['C-NAAF80'] = peaksAAPTM_332['C'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['D-NAAF80'] = peaksAAPTM_332['D'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['E-NAAF80'] = peaksAAPTM_332['E'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['F-NAAF80'] = peaksAAPTM_332['F'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['G-NAAF80'] = peaksAAPTM_332['G'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['H-NAAF80'] = peaksAAPTM_332['H'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['I-NAAF80'] = peaksAAPTM_332['I'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['K-NAAF80'] = peaksAAPTM_332['K'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['L-NAAF80'] = peaksAAPTM_332['L'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['M-NAAF80'] = peaksAAPTM_332['M'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['N-NAAF80'] = peaksAAPTM_332['N'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['P-NAAF80'] = peaksAAPTM_332['P'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['Q-NAAF80'] = peaksAAPTM_332['Q'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['R-NAAF80'] = peaksAAPTM_332['R'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['S-NAAF80'] = peaksAAPTM_332['S'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['T-NAAF80'] = peaksAAPTM_332['T'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['V-NAAF80'] = peaksAAPTM_332['V'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['W-NAAF80'] = peaksAAPTM_332['W'] * peaks332['NAAF factor']\n",
    "peaksAAPTM_332['Y-NAAF80'] = peaksAAPTM_332['Y'] * peaks332['NAAF factor']\n",
    "\n",
    "# multiply the NAAF80 factor by the PTMs normalize its abundance by peak area and peptide length\n",
    "\n",
    "peaksAAPTM_332['ccarb-NAAF80'] = peaksAAPTM_332['c-carb'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['moxid-NAAF80'] = peaksAAPTM_332['m-oxid'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['koxid-NAAF80'] = peaksAAPTM_332['k-oxid'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['poxid-NAAF80'] = peaksAAPTM_332['p-oxid'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['roxid-NAAF80'] = peaksAAPTM_332['r-oxid'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['yoxid-NAAF80'] = peaksAAPTM_332['y-oxid'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['ndeam-NAAF80'] = peaksAAPTM_332['n-deam'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['kmeth-NAAF80'] = peaksAAPTM_332['k-meth'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['rmeth-NAAF80'] = peaksAAPTM_332['r-meth'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['qpyro-NAAF80'] = peaksAAPTM_332['q-pyro'] * peaksAAPTM_332['NAAF factor']\n",
    "peaksAAPTM_332['kacet-NAAF80'] = peaksAAPTM_332['k-acet'] * peaksAAPTM_332['NAAF factor']\n",
    "\n",
    "# write the dataframe to a new csv\n",
    "peaksAAPTM_332.to_csv(\"/home/millieginty/Documents/git-repos/rot-mayer/data/processed/PeaksDN/TW_332_T12_undigested_combine_PTMopt_DN80_NAAF.csv\")\n",
    "\n",
    "peaksAAPTM_332.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# made a dataframe that's the sum of NAAF corrected AAs and PTMs\n",
    "\n",
    "index = ['sample total']\n",
    "\n",
    "data = {'NAAF': peaksAAPTM_332['NAAF factor'].sum(),\n",
    "        'A': peaksAAPTM_332['A-NAAF80'].sum(),\n",
    "        'C': peaksAAPTM_332['C-NAAF80'].sum(),\n",
    "        'D': peaksAAPTM_332['D-NAAF80'].sum(),\n",
    "        'E': peaksAAPTM_332['E-NAAF80'].sum(),\n",
    "        'F': peaksAAPTM_332['F-NAAF80'].sum(),\n",
    "        'G': peaksAAPTM_332['G-NAAF80'].sum(),\n",
    "        'H': peaksAAPTM_332['H-NAAF80'].sum(),\n",
    "        'I': peaksAAPTM_332['I-NAAF80'].sum(),\n",
    "        'K': peaksAAPTM_332['K-NAAF80'].sum(),\n",
    "        'L': peaksAAPTM_332['L-NAAF80'].sum(),\n",
    "        'M': peaksAAPTM_332['M-NAAF80'].sum(),\n",
    "        'N': peaksAAPTM_332['N-NAAF80'].sum(),\n",
    "        'P': peaksAAPTM_332['P-NAAF80'].sum(),\n",
    "        'Q': peaksAAPTM_332['Q-NAAF80'].sum(),\n",
    "        'R': peaksAAPTM_332['R-NAAF80'].sum(),\n",
    "        'S': peaksAAPTM_332['S-NAAF80'].sum(),\n",
    "        'T': peaksAAPTM_332['T-NAAF80'].sum(),\n",
    "        'V': peaksAAPTM_332['V-NAAF80'].sum(),\n",
    "        'W': peaksAAPTM_332['W-NAAF80'].sum(),\n",
    "        'Y': peaksAAPTM_332['Y-NAAF80'].sum(),\n",
    "        'c-carb': peaksAAPTM_332['ccarb-NAAF80'].sum(),\n",
    "        'm-oxid': peaksAAPTM_332['moxid-NAAF80'].sum(),\n",
    "        'k-oxid': peaksAAPTM_332['koxid-NAAF80'].sum(),\n",
    "        'p-oxid': peaksAAPTM_332['poxid-NAAF80'].sum(),\n",
    "        'r-oxid': peaksAAPTM_332['roxid-NAAF80'].sum(),\n",
    "        'y-oxid': peaksAAPTM_332['yoxid-NAAF80'].sum(),\n",
    "        'n-deam': peaksAAPTM_332['ndeam-NAAF80'].sum(),\n",
    "        'k-meth': peaksAAPTM_332['kmeth-NAAF80'].sum(),\n",
    "        'r-meth': peaksAAPTM_332['rmeth-NAAF80'].sum(),\n",
    "        'q-pyro': peaksAAPTM_332['qpyro-NAAF80'].sum(),\n",
    "        'k-acet': peaksAAPTM_332['kacet-NAAF80'].sum()\n",
    "       }\n",
    "\n",
    "totalpeaks80_NAAF = pd.DataFrame(data, columns=['NAAF', 'A', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'K', \\\n",
    "                                           'L', 'M', 'N', 'P', 'Q', 'R', 'S', 'T', 'V', \\\n",
    "                                           'W', 'Y', 'c-carb', 'm-oxid', 'k-oxid', 'p-oxid', \\\n",
    "                                           'r-oxid', 'y-oxid', 'n-deam', 'k-meth', 'r-meth', \\\n",
    "                                            'q-pyro', 'k-acet'], index=index)\n",
    "\n",
    "# calculate NAAF-corrected percentage of C's with carb (should be 1.0)\n",
    "totalpeaks80_NAAF['% C w/ carb.'] = totalpeaks80_NAAF['c-carb'] / totalpeaks80_NAAF['C'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of M's that are oxidized\n",
    "totalpeaks80_NAAF['% M w/ oxid'] = totalpeaks80_NAAF['m-oxid'] / totalpeaks80_NAAF['M'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of K's that are oxidized\n",
    "totalpeaks80_NAAF['% K w/ oxid'] = totalpeaks80_NAAF['k-oxid'] / totalpeaks80_NAAF['K'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of P's that are oxidized\n",
    "totalpeaks80_NAAF['% P w/ oxid'] = totalpeaks80_NAAF['p-oxid'] / totalpeaks80_NAAF['P'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of R's that are oxidized\n",
    "totalpeaks80_NAAF['% R w/ oxid'] = totalpeaks80_NAAF['r-oxid'] / totalpeaks80_NAAF['R'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of Y's that are oxidized\n",
    "totalpeaks80_NAAF['% Y w/ oxid'] = totalpeaks80_NAAF['y-oxid'] / totalpeaks80_NAAF['Y'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of N's that are deamidated\n",
    "totalpeaks80_NAAF['% N w/ deam'] = totalpeaks80_NAAF['n-deam'] / totalpeaks80_NAAF['N'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of K's that are methylated\n",
    "totalpeaks80_NAAF['% K w/ meth'] = totalpeaks80_NAAF['k-meth'] / totalpeaks80_NAAF['K'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of R's that are methylated\n",
    "totalpeaks80_NAAF['% R w/ meth'] = totalpeaks80_NAAF['r-meth'] / totalpeaks80_NAAF['R'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of Q's that are pyro glu'd\n",
    "totalpeaks80_NAAF['% Q w/ pyro'] = totalpeaks80_NAAF['q-pyro'] / totalpeaks80_NAAF['Q'] \n",
    "\n",
    "# calculate NAAF-corrected percentage of K's that are methylated\n",
    "totalpeaks80_NAAF['% K w/ acet'] = totalpeaks80_NAAF['k-acet'] / totalpeaks80_NAAF['K'] \n",
    "\n",
    "# calculate NAAF summed numerator over denominator (in above cell) for all peptides in dataset i: a check\n",
    "totalpeaks80_NAAF['NAAF check'] = totalpeaks80_NAAF['NAAF'] / 243531.178471\n",
    "\n",
    "# write modified dataframe to new txt file, same name + totals\n",
    "totalpeaks80_NAAF.to_csv(\"/home/millieginty/Documents/git-repos/rot-mayer/data/processed/PeaksDN/TW_332_T12_undigested_combine_PTMopt_DN80_NAAF_totals.csv\")\n",
    "\n",
    "totalpeaks80_NAAF.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export stripped peptides >80% ALC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# keep only stripped peptide column \n",
    "pep80 = peaks332[[\"stripped peptide\"]]\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "pep80.to_csv(\"/home/millieginty/Documents/git-repos/TW_332_T12_undigested_combine_PTMopt_DN80_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "# removing redundancy\n",
    "peaks80dedup = pd.DataFrame.drop_duplicates(pep80)\n",
    "\n",
    "# write altered dataframe to new txt file\n",
    "# used header and index parameters to get rid of 'Peptide' header and the indexing\n",
    "\n",
    "peaks80dedup.to_csv(\"/home/millieginty/Documents/git-repos/rot-mayer/data/processed/PeaksDN/TW_332_T12_undigested_combine_PTMopt_DN80_nonredundant_stripped_peptides.txt\", header=False, index=False)\n",
    "\n",
    "print(\"# redundant stripped Peaks peptides >80% ALC\", len(pep80))\n",
    "print(\"# nonredundant stripped Peaks peptides >80% ALC\", len(peaks80dedup))\n",
    "print(\"average peptide length Peaks peptides >80% ALC\", peaks332['stripped length'].mean())\n",
    "\n",
    "# count all unique peptide (modified peptides included)\n",
    "# keep only peptide column >80% ALC\n",
    "pep80m = peaks332[[\"Peptide\"]]\n",
    "\n",
    "# deduplicate\n",
    "pep80mdedup = pd.DataFrame.drop_duplicates(pep80m)\n",
    "\n",
    "print(\"# redundant Peaks peptides >80% ALC\", len(pep80m))\n",
    "print(\"# nonredundant Peaks peptides\", len(pep80mdedup))\n",
    "\n",
    "# check\n",
    "pep80.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
