{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PROTEUS2 protein secondary structure analysis:\n",
    "\n",
    "### This is a notebook to organize and evaluate the output of protein secondary structure prediction by [Proteus2](http://www.proteus2.ca/proteus2/), described in [Montgomerie et al., 2008](https://academic.oup.com/nar/article/36/suppl_2/W202/2506231) in _Nucleic Acids Research_. \n",
    "\n",
    "### The output of Proteus2 comes in an email in nominally FASTA format, but with extra line breaks and spacings within protein and prediction sequences. Also, the name gets cut off.\n",
    "\n",
    "#### They look like this:\n",
    "\n",
    "#### >Thalas\n",
    "\n",
    "MMKLAALAAL MGSAAAFAPA QTGKASTQLR AFEDELGAQP PLGFFDPFGM \n",
    "\n",
    "CCHHHHHHHH HHHHHCCCCC CCCCCCCCCC CCCCCCCCCC CCCCCCCCCC \n",
    "\n",
    "LSGDCTQERF DRLRYVEIKH GRICMLAFLG QIVTRAGIHL PGSINYAGDS \n",
    "\n",
    "CCCCCCHHHH HHHHHHHHHH HHHHHHHHHH HHHHHHHCCC CCCCCCCCCC \n",
    "\n",
    "FDSFPNGVAA LFGPNSIPTA GLVQIIAFIG VLECAFMRDV PGTGNEFVGD \n",
    "\n",
    "CCCCCCCCCC CCCCCCCCHH HHHHHHHHHH HHHHHHHHCC CCCCCCCCCC \n",
    "\n",
    "FRNGYIDFGW DDFDEETKLQ KRAIQSGTIS NMMKLAALAA LMGSAAAFAP \n",
    "\n",
    "CCCCCCCCCC CCCCHHHHHH HHHHHHHHHH HHHHHHHHHC CCCCCCCCCC \n",
    "\n",
    "\n",
    "\n",
    "### Output means:\n",
    "\n",
    "- H = Helix\n",
    "- E = Beta Strand\n",
    "- C = Coil\n",
    "- T = Membrane helix\n",
    "- B = Membrane strand\n",
    "- S = Signal peptide\n",
    "- c = Cleavage site"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /home/millieginty/Documents/git-repos/rot-mayer/analyses/proteus2/Proteins-Proteus2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'1.0.5'"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# LIBRARIES\n",
    "#import pandas library for working with tabular data\n",
    "import os\n",
    "os.getcwd()\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "from scipy.stats import kde\n",
    "#import regular expresson (regex)\n",
    "import re\n",
    "#check pandas version\n",
    "pd.__version__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PeaksDB_322_prot_trypsin_totals\r\n",
      "PeaksDB_323_prot_trypsin_totals\r\n",
      "PeaksDB_324_prot_trypsin_totals\r\n",
      "PeaksDB_325_prot_trypsin_totals\r\n",
      "PeaksDB_329_prot_undigested_totals\r\n",
      "PeaksDB_330_prot_undigested_totals\r\n",
      "T0_322_trypsin_PeaksDB_protein_proteus\r\n",
      "T0_322_trypsin_PeaksDB_protein_proteus_sort.csv\r\n",
      "T0_322_trypsin_PeaksDB_protein_proteus.txt\r\n",
      "T0_329_undigested_PeaksDB_protein_proteus\r\n",
      "T0_329_undigested_PeaksDB_protein_proteus_sort.csv\r\n",
      "T0_329_undigested_PeaksDB_protein_proteus.txt\r\n",
      "T12_325_trypsin_PeaksDB_protein_proteus\r\n",
      "T12_325_trypsin_PeaksDB_protein_proteus_sort.csv\r\n",
      "T12_325_trypsin_PeaksDB_protein_proteus.txt\r\n",
      "T12_332_undigested_PeaksDB_protein_proteus\r\n",
      "T2_323_trypsin_PeaksDB_protein_proteus\r\n",
      "T2_323_trypsin_PeaksDB_protein_proteus_sort.csv\r\n",
      "T2_323_trypsin_PeaksDB_protein_proteus.txt\r\n",
      "T2_330_undigested_PeaksDB_protein_proteus\r\n",
      "T2_330_undigested_PeaksDB_protein_proteus_sort.csv\r\n",
      "T2_330_undigested_PeaksDB_protein_proteus.txt\r\n",
      "T5_324_trypsin_PeaksDB_protein_proteus\r\n",
      "T5_324_trypsin_PeaksDB_protein_proteus_sort.csv\r\n",
      "T5_324_trypsin_PeaksDB_protein_proteus.txt\r\n",
      "T5_331_undigested_PeaksDB_protein_proteus\r\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "Skip to content\r\n",
      "Using UW Mail with screen readers\r\n",
      "Meet\r\n",
      "Hangouts\r\n",
      "1 of 16,525\r\n",
      "Proteus 2.0 done\r\n",
      "Inbox\r\n",
      "proteus2@wishartlab.com\r\n",
      "\t\r\n"
     ]
    }
   ],
   "source": [
    "!head T5_331_undigested_PeaksDB_protein_proteus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove empty lines\n",
    "# remove empty spaces from lines\n",
    "# remove carat protein sequence names\n",
    "# collapse the 2 lines from every protein sequence line (protein seq, secondard stucture pred.) into one line\n",
    "\n",
    "!sed '/^[[:space:]]*$/d' T5_331_undigested_PeaksDB_protein_proteus | cat \\\n",
    "| sed '/>/d' \\\n",
    "| tr -d \"[:blank:]\" > T5_331_undigested_PeaksDB_protein_proteus.txt\n",
    "\n",
    "!awk '{printf \"%s%s\",$0,(NR%2?FS:RS)}' T5_331_undigested_PeaksDB_protein_proteus.txt > \\\n",
    "T5_331_undigested_PeaksDB_protein_proteus_sort.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Skiptocontent UsingUWMailwithscreenreaders\r\n",
      "Meet Hangouts\r\n",
      "1of16,525 Proteus2.0done\r\n",
      "Inbox proteus2@wishartlab.com\r\n",
      "Mon,Apr5,12:23PM(1dayago) THANKYOUFORUSINGPROTEUS2************************************************************PROTEUS2isoperatedandmaintainedbytheGenomeCanadaBioinformatic\r\n",
      "13 proteus2@wishartlab.com\r\n",
      "5:48PM(1hourago) ******************************Numberofsequencessubmitted:29Numberofsequencessuccessfullyprocessed:29Timeofsubmission:06:09:02Apr06,2021Timeo\r\n",
      "KMKLAILAALFGSAAAFAPSQTGKASTQLRAFEDELGAQPPLGFFDPFGM CCCCHHHHHHHHHHCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC\r\n",
      "LSGDCTQERFDRLRYVEIKHGRICMLAFLGQVVTRAGIHLPGSINYAGDS CCCCCCHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHCCCCCCCCCCC\r\n",
      "FDSFPNGVAALFGPNSIPTAGLVQIIAFIGVLECAFMRDVPGTGNEFVGD CCCCCCCCCCCCCCCCCHHHHHHHHHHHHHHHHHHHHHCCCCCCCCCCCC\r\n"
     ]
    }
   ],
   "source": [
    "!head T5_331_undigested_PeaksDB_protein_proteus_sort.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read into pandas with space delimeter\n",
    "\n",
    "PeaksDB_331_prot = pd.read_csv(\"T5_331_undigested_PeaksDB_protein_proteus_sort.csv\", delim_whitespace=True, header=None)\n",
    "\n",
    "\n",
    "# name columns\n",
    "\n",
    "PeaksDB_331_prot.columns =['Stripped protein sequence', 'Secondary structure pred.'] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Stripped protein sequence</th>\n",
       "      <th>Secondary structure pred.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Skiptocontent</td>\n",
       "      <td>UsingUWMailwithscreenreaders</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Meet</td>\n",
       "      <td>Hangouts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1of16,525</td>\n",
       "      <td>Proteus2.0done</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Inbox</td>\n",
       "      <td>proteus2@wishartlab.com</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mon,Apr5,12:23PM(1dayago)</td>\n",
       "      <td>THANKYOUFORUSINGPROTEUS2**********************...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Stripped protein sequence  \\\n",
       "0              Skiptocontent   \n",
       "1                       Meet   \n",
       "2                  1of16,525   \n",
       "3                      Inbox   \n",
       "4  Mon,Apr5,12:23PM(1dayago)   \n",
       "\n",
       "                           Secondary structure pred.  \n",
       "0                       UsingUWMailwithscreenreaders  \n",
       "1                                           Hangouts  \n",
       "2                                     Proteus2.0done  \n",
       "3                            proteus2@wishartlab.com  \n",
       "4  THANKYOUFORUSINGPROTEUS2**********************...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PeaksDB_331_prot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# add a column with the stripped peptide length (number of AAs)\n",
    "PeaksDB_331_prot['Sequence length'] = PeaksDB_331_prot['Stripped protein sequence'].apply(len)\n",
    "\n",
    "# use a count function to enumerate the # of C's (coil residues) for each peptide\n",
    "PeaksDB_331_prot['C'] = PeaksDB_331_prot['Secondary structure pred.'].str.count(\"C\")\n",
    "\n",
    "# use a count function to enumerate the # of H's (helices residues) in each peptide\n",
    "PeaksDB_331_prot['H'] = PeaksDB_331_prot['Secondary structure pred.'].str.count(\"H\")\n",
    "\n",
    "# use a count function to enumerate the # of E's (beta strand residues) in each peptide\n",
    "PeaksDB_331_prot['E'] = PeaksDB_331_prot['Secondary structure pred.'].str.count(\"E\")\n",
    "\n",
    "#use a count function to enumerate the # of T's (membrane helix residues) in each peptide\n",
    "PeaksDB_331_prot['T'] = PeaksDB_331_prot['Secondary structure pred.'].str.count(\"T\")\n",
    "\n",
    "# use a count function to enumerate the # of B's (membrane strand residues) in each peptide\n",
    "PeaksDB_331_prot['B'] = PeaksDB_331_prot['Secondary structure pred.'].str.count(\"B\")\n",
    "\n",
    "# use a count function to enumerate the # of B's (signal peptide residues) in each peptide\n",
    "PeaksDB_331_prot['S'] = PeaksDB_331_prot['Secondary structure pred.'].str.count(\"S\")\n",
    "\n",
    "# use a count function to enumerate the # of c's (cleavage site residues) in each peptide\n",
    "PeaksDB_331_prot['c'] = PeaksDB_331_prot['Secondary structure pred.'].str.count(\"c\")\n",
    "\n",
    "# add a column with the % C\n",
    "PeaksDB_331_prot['% C'] = PeaksDB_331_prot['C'] / PeaksDB_331_prot['Sequence length']\n",
    "\n",
    "# add a column with the % H\n",
    "PeaksDB_331_prot['% H'] = PeaksDB_331_prot['H'] / PeaksDB_331_prot['Sequence length']\n",
    "\n",
    "# add a column with the % E\n",
    "PeaksDB_331_prot['% E'] = PeaksDB_331_prot['E'] / PeaksDB_331_prot['Sequence length']\n",
    "\n",
    "# add a column with the % T\n",
    "PeaksDB_331_prot['% T'] = PeaksDB_331_prot['T'] / PeaksDB_331_prot['Sequence length']\n",
    "\n",
    "# add a column with the % B\n",
    "PeaksDB_331_prot['% B'] = PeaksDB_331_prot['B'] / PeaksDB_331_prot['Sequence length']\n",
    "\n",
    "# add a column with the % S\n",
    "PeaksDB_331_prot['% S'] = PeaksDB_331_prot['S'] / PeaksDB_331_prot['Sequence length']\n",
    "\n",
    "# add a column with the % c\n",
    "PeaksDB_331_prot['% c'] = PeaksDB_331_prot['c'] / PeaksDB_331_prot['Sequence length']\n",
    "\n",
    "# additive check\n",
    "\n",
    "PeaksDB_331_prot['% check'] = PeaksDB_331_prot['% C'] + PeaksDB_331_prot['% H'] + PeaksDB_331_prot['% E'] \\\n",
    "                                + PeaksDB_331_prot['% T'] + PeaksDB_331_prot['% B'] + \\\n",
    "                                PeaksDB_331_prot['% S'] + PeaksDB_331_prot['% c']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Stripped protein sequence</th>\n",
       "      <th>Secondary structure pred.</th>\n",
       "      <th>Sequence length</th>\n",
       "      <th>C</th>\n",
       "      <th>H</th>\n",
       "      <th>E</th>\n",
       "      <th>T</th>\n",
       "      <th>B</th>\n",
       "      <th>S</th>\n",
       "      <th>c</th>\n",
       "      <th>% C</th>\n",
       "      <th>% H</th>\n",
       "      <th>% E</th>\n",
       "      <th>% T</th>\n",
       "      <th>% B</th>\n",
       "      <th>% S</th>\n",
       "      <th>% c</th>\n",
       "      <th>% check</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Skiptocontent</td>\n",
       "      <td>UsingUWMailwithscreenreaders</td>\n",
       "      <td>13</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.076923</td>\n",
       "      <td>0.076923</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Meet</td>\n",
       "      <td>Hangouts</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.250000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1of16,525</td>\n",
       "      <td>Proteus2.0done</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Inbox</td>\n",
       "      <td>proteus2@wishartlab.com</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>0.200000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Mon,Apr5,12:23PM(1dayago)</td>\n",
       "      <td>THANKYOUFORUSINGPROTEUS2**********************...</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.08</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.12</td>\n",
       "      <td>0.040000</td>\n",
       "      <td>0.480000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Stripped protein sequence  \\\n",
       "0              Skiptocontent   \n",
       "1                       Meet   \n",
       "2                  1of16,525   \n",
       "3                      Inbox   \n",
       "4  Mon,Apr5,12:23PM(1dayago)   \n",
       "\n",
       "                           Secondary structure pred.  Sequence length  C  H  \\\n",
       "0                       UsingUWMailwithscreenreaders               13  0  0   \n",
       "1                                           Hangouts                4  0  1   \n",
       "2                                     Proteus2.0done                9  0  0   \n",
       "3                            proteus2@wishartlab.com                5  0  0   \n",
       "4  THANKYOUFORUSINGPROTEUS2**********************...               25  1  1   \n",
       "\n",
       "   E  T  B  S  c   % C   % H   % E   % T   % B   % S       % c   % check  \n",
       "0  0  0  0  0  1  0.00  0.00  0.00  0.00  0.00  0.00  0.076923  0.076923  \n",
       "1  0  0  0  0  0  0.00  0.25  0.00  0.00  0.00  0.00  0.000000  0.250000  \n",
       "2  0  0  0  0  0  0.00  0.00  0.00  0.00  0.00  0.00  0.000000  0.000000  \n",
       "3  0  0  0  0  1  0.00  0.00  0.00  0.00  0.00  0.00  0.200000  0.200000  \n",
       "4  2  3  1  3  1  0.04  0.04  0.08  0.12  0.04  0.12  0.040000  0.480000  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "PeaksDB_331_prot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>% C total</th>\n",
       "      <th>% H total</th>\n",
       "      <th>% E total</th>\n",
       "      <th>% T total</th>\n",
       "      <th>% B total</th>\n",
       "      <th>% S total</th>\n",
       "      <th>% c total</th>\n",
       "      <th>% check sum</th>\n",
       "      <th>overall % sum</th>\n",
       "      <th>overall % C</th>\n",
       "      <th>overall % H</th>\n",
       "      <th>overall % E</th>\n",
       "      <th>overall % T</th>\n",
       "      <th>overall % B</th>\n",
       "      <th>overall % S</th>\n",
       "      <th>overall % c</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>331 total</th>\n",
       "      <td>147.460209</td>\n",
       "      <td>80.845948</td>\n",
       "      <td>46.137316</td>\n",
       "      <td>30.211526</td>\n",
       "      <td>0.04</td>\n",
       "      <td>0.12</td>\n",
       "      <td>1.129423</td>\n",
       "      <td>305.944423</td>\n",
       "      <td>305.944423</td>\n",
       "      <td>0.481984</td>\n",
       "      <td>0.26425</td>\n",
       "      <td>0.150803</td>\n",
       "      <td>0.098748</td>\n",
       "      <td>0.000131</td>\n",
       "      <td>0.000392</td>\n",
       "      <td>0.003692</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            % C total  % H total  % E total  % T total  % B total  % S total  \\\n",
       "331 total  147.460209  80.845948  46.137316  30.211526       0.04       0.12   \n",
       "\n",
       "           % c total  % check sum  overall % sum  overall % C  overall % H  \\\n",
       "331 total   1.129423   305.944423     305.944423     0.481984      0.26425   \n",
       "\n",
       "           overall % E  overall % T  overall % B  overall % S  overall % c  \n",
       "331 total     0.150803     0.098748     0.000131     0.000392     0.003692  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index = ['331 total']\n",
    "\n",
    "data = {\n",
    "        '% C total': PeaksDB_331_prot['% C'].sum(),\n",
    "        '% H total': PeaksDB_331_prot['% H'].sum(),\n",
    "        '% E total': PeaksDB_331_prot['% E'].sum(),\n",
    "        '% T total': PeaksDB_331_prot['% T'].sum(),\n",
    "        '% B total': PeaksDB_331_prot['% B'].sum(),\n",
    "        '% S total': PeaksDB_331_prot['% S'].sum(),\n",
    "        '% c total': PeaksDB_331_prot['% c'].sum(),\n",
    "        '% check sum': PeaksDB_331_prot['% check'].sum()\n",
    "       }\n",
    "\n",
    "PeaksDB_331_prot_totals = pd.DataFrame(data, columns=['% C total', '% H total', '% E total', '% T total', \\\n",
    "                                                      '% B total',  '% S total', '% c total', \\\n",
    "                                                      '% check sum'], index=index)\n",
    "\n",
    "PeaksDB_331_prot_totals['overall % sum'] = PeaksDB_331_prot_totals['% C total'] \\\n",
    "                                            + PeaksDB_331_prot_totals['% H total'] \\\n",
    "                                            + PeaksDB_331_prot_totals['% E total'] \\\n",
    "                                            + PeaksDB_331_prot_totals['% T total'] \\\n",
    "                                            + PeaksDB_331_prot_totals['% B total'] \\\n",
    "                                            + PeaksDB_331_prot_totals['% S total'] \\\n",
    "                                            + PeaksDB_331_prot_totals['% c total'] \n",
    "\n",
    "\n",
    "PeaksDB_331_prot_totals['overall % C'] = PeaksDB_331_prot_totals['% C total'] / PeaksDB_331_prot_totals['overall % sum']\n",
    "\n",
    "PeaksDB_331_prot_totals['overall % H'] = PeaksDB_331_prot_totals['% H total'] / PeaksDB_331_prot_totals['overall % sum']\n",
    "\n",
    "PeaksDB_331_prot_totals['overall % E'] = PeaksDB_331_prot_totals['% E total'] / PeaksDB_331_prot_totals['overall % sum']\n",
    "\n",
    "PeaksDB_331_prot_totals['overall % T'] = PeaksDB_331_prot_totals['% T total'] / PeaksDB_331_prot_totals['overall % sum']\n",
    "\n",
    "PeaksDB_331_prot_totals['overall % B'] = PeaksDB_331_prot_totals['% B total'] / PeaksDB_331_prot_totals['overall % sum']\n",
    "\n",
    "PeaksDB_331_prot_totals['overall % S'] = PeaksDB_331_prot_totals['% S total'] / PeaksDB_331_prot_totals['overall % sum']\n",
    "\n",
    "PeaksDB_331_prot_totals['overall % c'] = PeaksDB_331_prot_totals['% c total'] / PeaksDB_331_prot_totals['overall % sum']\n",
    "\n",
    "# write to csv\n",
    "\n",
    "PeaksDB_331_prot_totals.to_csv(\"/home/millieginty/Documents/git-repos/rot-mayer/analyses/proteus2/Proteins-Proteus2/PeaksDB_331_prot_undigested_totals\")\n",
    "\n",
    "PeaksDB_331_prot_totals.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
